### PR TITLE
Single channel should show as black and white in viewport

### PR DESCRIPTION
**Pull request by JGamache-autodesk**
A single channel image would show as red/black when used in a RGB context. Make sure those images show a black and white.

### CODE REVIEW

File: test/testSamples/MaterialX/textures/R16F.exr
Comment (JGamache-autodesk) [inline test/testSamples/MaterialX/textures/R16F.exr:1]: Created test textures in 16/32 bit EXR with either Mono or Mono+Alpha channels to exercise all conversions.

File: test/lib/mayaUsd/render/vp2RenderDelegate/VP2RenderDelegateMaterialXTest/baseline/DemoQuads_render.png
Comment (JGamache-autodesk) [inline test/lib/mayaUsd/render/vp2RenderDelegate/VP2RenderDelegateMaterialXTest/baseline/DemoQuads_render.png:1]: Tests all possible Mono and Mono+Alpha precisions.

File: lib/mayaUsd/render/vp2RenderDelegate/material.cpp [hunk 1]
```diff
@@ -1488,24 +1488,107 @@ MHWRender::MTexture* _LoadTexture(
...
+        desc.fFormat = MHWRender::kR32G32B32_FLOAT;
+        desc.fBytesPerRow = spec.width * bpp_RGB32;
+        desc.fBytesPerSlice = desc.fBytesPerRow * spec.height;
+
+        std::vector<unsigned char> texels(desc.fBytesPerSlice);
+
+        for (int y = 0; y < spec.height; y++) {
+            for (int x = 0; x < spec.width; x++) {
+                const int t = spec.width * y + x;
+                for (int b = 0; b < bpp_RGB32; b++) {
```
Comment (vlasovi) [inline lib/mayaUsd/render/vp2RenderDelegate/material.cpp:1504]: This is a time-critical cycle. Do we want to multithread it or optimize otherwise?

**Review by vlasovi — DISMISSED** [review]
Looks good. The only thing is this change adds a texture format conversion cycle for some texture types, which might slow down a bit texture loading. I don't think this is a big concern, though.

File: lib/mayaUsd/render/vp2RenderDelegate/material.cpp [hunk 2]
```diff
@@ -1501,8 +1503,11 @@ MHWRender::MTexture* _LoadTexture(
         for (int y = 0; y < spec.height; y++) {
             for (int x = 0; x < spec.width; x++) {
                 const int t = spec.width * y + x;
-                for (int b = 0; b < bpp_RGB32; b++) {
-                    texels[t * bpp_RGB32 + b] = storage[t * bpp + (b % 4)];
+                for (int b = 0; b < bits_RGB32; b++) {
+                    const unsigned char bits = storage[t * bpp + b];
```
Comment (JGamache-autodesk) [inline lib/mayaUsd/render/vp2RenderDelegate/material.cpp:1507]: Read the source buffer only once.

**Review by vlasovi — APPROVED** [review]

**Review by vlasovi — APPROVED** [review]
Nice. This looks very good now.

**Review by vlasovi — APPROVED** [review]

**Review by vlasovi — APPROVED** [review]

**Comment by JGamache-autodesk** [comment]
@seando-adsk ready for merge. Thanks!

**Pull request merged**